src/axom/core/NumericLimits.hpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ *
+ * \file NumericLimits.hpp
+ *
+ * \brief Header file containing portability layer for std::numeric_limits
+ *        capabilities
+ *
+ */
+
+#ifndef AXOM_NUMERICLIMITS_HPP_
+#define AXOM_NUMERICLIMITS_HPP_
+
+#include "axom/config.hpp"  // for compile-time definitions
+
+#include <limits>
+
+#if defined(AXOM_USE_CUDA)
+  #include <cuda/std/limits>
+#endif
+
+namespace axom
+{
+#if defined(AXOM_USE_CUDA) && defined(AXOM_DEVICE_CODE)
```

Comment:
Would it make sense to remove `if defined(AXOM_DEVICE_CODE)` and just have the `AXOM_USE_CUDA` guard? It's my impression that `cuda::std` should work on both the host and device.

Reply:
If it does, that would be preferable. I need to look into that. @kennyweiss discussed this PR yesterday. That resulted in some concerns about several things in the code that we should discuss as a team.  Unfortunately, that will have to wait for a couple of weeks as next Monday is a LLNL holiday and NECDC is the week after that.

Reply:
@publixsubfan `cuda::std` does work in device and host code. I ran into some issues with some Axom tests where `cuda::std::numeric_limits` does not support `long double`. The intent of my change was to use `std::numeric_limits` in host code for all builds. However, it's not clear to me that we need to support `long double`.